### ISSUE TITLE
Migrate entire project to use SafeArgs in fragments instead of raw bundles

### ISSUE THREAD

**Issue by haroldadmin** [fossasia/open-event-android#1205]
**Describe the bug**
Large parts of the app currently use raw bundles to access and pass arguments. This is unsafe and does not utilize the Navigation Architecture Component to its fullest extent.

**Additional context**
Since we already use the Navigation architecture component, it is easier to migrate to using SafeArgs. This will also be benficial if we decide to use Actions in the Nav AAC to switch between different screens.

**Would you like to work on the issue?**
- [x] Yes

**Comment by haroldadmin** [comment]
@iamareebjamal @liveHarshit Do you think this is worth pursuing? 
I want to do this fragment by fragment, but I thought opening a new issue for every fragment is spamming.

**Comment by liveHarshit** [comment]
> Do you think this is worth pursuing?

Go ahead.

**Comment by haroldadmin** [comment]
Okay, thanks!

**Comment by haroldadmin** [comment]
I have completed the refactor of the entire project to safe args, and I want to do one more thing that would make things a lot better.

Earlier most of the keys used to retrieve data from fragment arguments were stored as completely capitalized `const val` fields at the top of each fragment. But now these are obsolete, because we don't need keys to access fragment args anymore. For example, if a fragment used `const val EVENT_ID = "EVENT_ID"` as a key, i added an argument to the nav graph with the name `EVENT_ID`. 

There is a slight readability problem because of such completely capitalized arg keys, though. Now the SafeArgs gradle plugin generates classes with methods named like `setEVENTID(...)` etc.

I want to refactor the argument names to be in camel case so that they feel natural when accessed through kotlin code. I also want to put all their names in the strings.xml file (or another xml file called arguments.xml).

Should I go ahead and do this step too? @iamareebjamal @liveHarshit

**Comment by haroldadmin** [comment]
Another quick question: Will multiple commits for this PR be okay? I currently have one commit for each fragment that I have refactored. Should I squash them?

**Comment by liveHarshit** [comment]
> because we don't need keys to access fragment args anymore.

Then remove these keys, if not used.

**Comment by liveHarshit** [comment]
> Another quick question: Will multiple commits for this PR be okay? I currently have one commit for each fragment that I have refactored. Should I squash them?

You can squash them later when PR is approved.

**Comment by haroldadmin** [comment]
> > Another quick question: Will multiple commits for this PR be okay? I currently have one commit for each fragment that I have refactored. Should I squash them?
> 
> You can squash them later when PR is approved.

Cool. That works too. Multiple commits will make them easier to review as well.

**Comment by haroldadmin** [comment]
> > because we don't need keys to access fragment args anymore.
> 
> Then remove these keys, if not used.

I also want to rename the arguments so that they feel natural when used in Kotlin/Java code. For example, change the name of EVENT_ID to eventId. Would that be fine?

**Comment by liveHarshit** [comment]
> And is moving these arg names to strings.xml a good idea?

IMO, not a good idea for getting these from string resources. And use as constant.

**Comment by liveHarshit** [comment]
String resources should use for layout purposes.

**Comment by haroldadmin** [comment]
Alright. I'm waiting on @iamareebjamal's review for the EventDetailsFragment safe args PR before I submit one for this issue.
